src/types.rs

```diff
@@ -1,12 +1,13 @@
-use std::{error::Error, fmt::Debug};
+use std::fmt::Debug;
 
 use tokio::sync::mpsc;
 use tracing::{info, warn};
 
-#[derive(Copy, Debug, Clone)]
-pub enum ScannerMessage<T: Clone, E: Error + Clone> {
+use crate::ScannerError;
+
+#[derive(Debug, Clone)]
+pub enum ScannerMessage<T: Clone> {
     Data(T),
-    Error(E),
     Notification(Notification),
 }
```

Comment:
```suggestion
    Notification(Notification),
}

pub type ScannerResult<T: Clone> = Result<ScannerMessage<T>, ScannerError>;
```
tip: define a convenience type both for us and scanner users